pgsqltoolsservice/workspace/workspace.py

```diff
@@ -67,6 +67,7 @@ def open_file(self, file_uri: str, initial_buffer: Optional[str] = None) -> Opti
 
         # If the file is already loaded in the workspace, just return it
         script_file: ScriptFile = self._workspace_files.get(file_uri)
+        file_uri = unquote(file_uri)
```

Comment:
Don't you want to do this before checking the existing workspace files? Otherwise say I open a file whose URI is passed in as 

file://c%3a/my/file.txt

It's then unquoted to 

file://c:/my/file.txt

and added to _workspace_files.

Then another request comes in and this function is called again with the URI

file://c%3a/my/file.txt

which it again fails to find.

Reply:
@Charles-Gagnon are you asking about a scenario of opening the same file again ? sorry I dont understand this.

Reply:
I basically have tested scenarios lile:
1. Open new query file and paste query
2. Open saved file and execute query 
3. try to open the same file again and execute the query 
4. Close ADS and run queries from files that are already opened from previous query

Reply:
Hm, interesting. And did you verify whether subsequent calls to open_file are correctly retrieving the existing file (instead of making another one?) I'm just seeing this function called in a whole bunch of places and so if those are all passing in the escaped URI then I'm surprised it's not failing for

Reply:
@Charles-Gagnon, I get what you were saying. I did miss few other changes as part of this. Now I have fixed it. Thanks for catching